### PR TITLE
fix: repo/owner parsing for gitlab

### LINKED ISSUE #4247: GitLab subgroups broken?
### Component

server

### Describe the bug

Just configured a new instance against gitlab.com.

Facing the following issues:

- Unable to start builds due to `[generic]could not load config from forge: %!w(<nil>)`

Server logs report:

```
/server/services/config/forge.go:150","message":"could not get folder from forge"}
```

The repo exists in a subgroup and hast just been enabled.

- Subgroups are sometimes visualized in the UI dashboard, sometimes not
	- Enabled two repos in two different subgroups: one was listed as `group/subgroup/repo`, the other one as `group/repo`

Maybe related: #2311

### Steps to reproduce

.

### Expected behavior

_No response_

### System Info

```shell
Gitlab 17.6
Woodpecker next `next-9dd796ad65` (Oct 13th 2024)
WP 2.7.1
```

### Additional context

_No response_

### Validations

- [X] Read the [docs](https://woodpecker-ci.org/docs/intro).
- [X] Check that there isn't [already an issue](https://github.com/woodpecker-ci/woodpecker/issues) that reports the same bug to avoid creating a duplicate.
- [X] Checked that the bug isn't fixed in the `next` version already [https://woodpecker-ci.org/faq#which-version-of-woodpecker-should-i-use]

### PR DESCRIPTION
**Pull request by pat-s**
`extractFromPath` wrongly splitted owner/group/repo patterns from repos in subgroups. 

Before, `owner/group/repo` resulted in `owner` (for owner) and `repo` for repo. However, it should be `owner/group` for owner. Otherwise, webhooks for repos with subpaths fail.

There was no test before, hence I added one. I want to test this first with PR images.

fix #4247

### CODE REVIEW

File: server/forge/gitlab/gitlab_test.go [hunk 1]
```diff
@@ -300,3 +300,17 @@ func Test_GitLab(t *testing.T) {
 		})
 	})
 }
+
+// Test function
```
Comment (6543) [inline server/forge/gitlab/gitlab_test.go:304]:
```suggestion
// Test extractFromPath function.
```

**Comment by pat-s** [comment]
Confirmed working.

**Review by anbraten — APPROVED** [review]
Code lgtm

**Comment by 6543** [comment]
backport as it's a bug?

**Comment by 6543** [comment]
->  #4261

**Pull request merged**